test/ecto/query/builder/limit_offset_test.exs

```diff
@@ -10,4 +10,16 @@ defmodule Ecto.Query.Builder.LimitOffsetTest do
     query = "posts" |> offset([], 1) |> offset([], 2) |> select([], 3)
     assert query.offset.expr == 2
   end
+
+  test "does not allow query variables in limit and offset" do
+    assert_raise Ecto.Query.CompileError, "query variables are not allowed in limit expression", fn ->
+      quoted = quote do: from p in "posts", limit: p.x + 1
+      Code.eval_quoted(quoted, [], __ENV__)
```

Comment:
I did it like this because the helper here didn't quite work for this case: https://github.com/elixir-ecto/ecto/blob/master/test/support/eval_helpers.exs.

It seems like the query compiled before going into the macro.

Reply:
actually i didn't import the macro...gg